Adds "Closing the Support Information Gap" lab (#277)
replicatedhq/kots-field-labs: instruqt/closing-information-gap/02-working-with-support-bundles/assignment.md

type: challenge
title: Working with Support Bundles
teaser: |-
  Learn how support bundles help you understand what is going
  on with a customer instance
notes:
- type: text
  contents: Let's learn about Support Bundles
tabs:
- title: Shell
  type: terminal
  hostname: shell
- title: Manifest Editor
  type: code
  hostname: shell
  path: /home/replicant
difficulty: basic
timelimit: 420
---

Instance Insights is a powerful tool to help you understand information you
need to know to assist a customer. It's the first tool you should pick up to
get context for the issue and can lead you toward the right troubleshooting
path. It was built to provide a set of insights for any application distributed
with the Replicated application. This limits it to information that's common
across applications.

Support bundles are another critical tool for troubleshooting customer issues.
The Replicated Platform allows you to define a bundle that your customer can
send to give you even more visibility into their environment. Since you define
the support bundle, it can collect details that are specific to troubleshooting
your application. Support bundles can also surface specific issues and provide

> guidance to your customer in order to repair issues on their own. They are part

Author: I don't love either option now, going to opt for "resolve"

of the [Troubleshoot](https://troubleshoot.sh) open source project.

What is a Support Bundle?
=========================

Support Bundles collect the information you need to understand your customer's
cluster and how your application is running in it. The Replicated Platform
allows you to do this without installing anything else into the cluster.

You define your bundle in a YAML file that follows the same format as a
Kubernetes object. The simplest support bundle object looks like this, and it's

> in the file `simplest-support-bundle.yaml`:

Author: 👍

```
apiVersion: troubleshoot.sh/v1beta2
kind: SupportBundle
metadata:
  name: simplest-support-bundle
spec:
  collectors: []
  analyzers: []
```

Let's try it out.

```
kubectl support-bundle ./simplest-support-bundle.yaml
```

It will take a few seconds to generate a support bundle in a file named
`support-bundle-$TIMESTAMP.tar.gz` that contains some simplest information
about the cluster. You can get a flavor for what's in the bundle by running

```
tar -tzf support-bundle-*.tar.gz | less
```

You'll see the files that were collected cataloging all of the resources in the
cluster and some information about the cluster itself. If you completed the
[Avoiding Installation
Pitfalls](https://play.instruqt.com/replicated/tracks/avoiding-installation-pitfalls)
lab this may surprise you, since a preflight check definition with an empty
`analzers` array caused an error. This is because the bundle is valid if it
only collects information, and by default both preflight checks and support

> bundles will collect a minimum set of information.

Author: That might be a log in thing…we're running the labs using "kiosk mode" which means you don't need to log in, but this link doesn't pass the required token along.

---
slug: working-with-support-bundles
id: gejrijpn0r1a
type: challenge
title: Working with Support Bundles
teaser: |-
  Learn how support bundles help you understand what is going
  on with a customer instance
notes:
- type: text
  contents: Let's learn about Support Bundles
tabs:
- title: Shell
  type: terminal
  hostname: shell
- title: Manifest Editor
  type: code
  hostname: shell
  path: /home/replicant
difficulty: basic
timelimit: 420
---

Instance Insights is a powerful tool to help you understand information you
need to know to assist a customer. It's the first tool you should pick up to
get context for the issue and can lead you toward the right troubleshooting
path. It was built to provide a set of insights for any application distributed
with the Replicated application. This limits it to information that's common
across applications.

Support bundles are another crticial tool for troubleshooting customer issues.
The Replicated Platform allows you to define a bundle that your customer can

> send to give you even more visibility inot their environment. Since you define

Reviewer: ```suggestion
send to give you even more visibility into their environment. Since you define
```

the support bundle, it can collect details that are specific to troubleshooting
your application. Support bundles can also surface specific issues and provide
guidance to your customer in order to repair issues on their own. They are part
of the [Troubleshoot](https://troubleshoot.sh) open source project.

What is a Support Bundle?
=========================

> Support Bundles collect the information you need to understand their

Reviewer: ```suggestion
Support Bundles collect the information you need to understand a customer's
```

cluster and how your application is running in it. The Replicated
Platform allows you to do this without installing anything else

> into the cluster

Reviewer: ```suggestion
into the cluster.
```

      - pass:
          message: |
            Ther Harbor core workload is running on this cluster and ready for use.
```

Taken together, your support bundle definition will look like
this:

```
apiVersion: troubleshoot.sh/v1beta2
kind: SupportBundle
metadata:
  name: harbor-support-bundle
spec:
  collectors:
    - logs:
        name: /app/harbor/logs
        selector:
          - app.kubernetes.io/name=harbor
        selector:
          - app=harbor
  analyzers:
    - deploymentStatus:
        name: harbor-core
        namespace: default
        outcomes:
          - fail:
              when: "absent"
              message: |
                The Harbor core workload has not been deployed to this cluster. Please sure to install the Harbor registry application using its Helm chart.
          - fail:
              when: "< 1"
              message: |
                The Harbor core workload is not currently running on this cluster. Please review the logs in this support bundle to locate any errors.
          - pass:
              message: |
                Ther Harbor core workload is running on this cluster and ready for use.
```

Getting Started
===============

> Let's creating a support bundle using this definition. Click on the

Reviewer: ```suggestion
Let's create a support bundle using this definition. Click on the
```
